### ISSUE TITLE
Re-implement "recurse" action

### ISSUE THREAD

**Issue by purpleKarrot** [ryppl/Boost2Git#3]
The new DSL has no "recourse" action yet.
Either it should get one, or we provide an additional file with all the directories listed that should be recoursed into. It may also be helpful to limit the version range where this rule should be considered.

**Comment by dabrahams** [comment]
Another alternative would be to keep track of enough information to know when recursion is needed.  I suppose that can't work if you start from the middle of the log, though…

**Comment by purpleKarrot** [comment]
Assuming we have two rules "/foo/bar/cow/" and "/foo/". When the directory "/foo/bar/" is copied, "/foo/bar/cow/" is not in the changelog. The change will match "/foo/". Recursion is required.

We can be sure that, whenever "/foo/" is matched, we should recurse. We know that, because there is another rule that begins with "/foo/".

It would be possible to flag all rules that require recursion by analyzing whether there is another rule that begins with that rule.

I tried that (but only whith linear search). Since we have about 400000 rules, it takes quite some time. I disabled it again and extended the DSL insted. Now a '?' after the match string causes the rule to recurse.
